lms/data_tasks/report/create_from_scratch/02_entities/01_users/03_users_sensitive/01_create_view.sql

```diff
@@ -0,0 +1,30 @@
+DROP MATERIALIZED VIEW IF EXISTS report.users_sensitive CASCADE;
+
+-- Create a separate table for sensitive user information, which should not be
+-- stored out of region.
+CREATE MATERIALIZED VIEW report.users_sensitive AS (
+    SELECT DISTINCT
+        user_map.user_id AS id,
+
+        -- The order here should ensure empty strings and nulls are sorted to
+        -- the back, so the "first" value should be non-null if possible
+        FIRST_VALUE(users.email) OVER (
+            PARTITION BY user_map.user_id
+            ORDER BY users.email
+            RANGE BETWEEN
+                UNBOUNDED PRECEDING AND
+                UNBOUNDED FOLLOWING
+        ) AS email,
```

Comment:
This is kind of a lot, but does seem to work. I don't know why sorting ASC (which is the default) doesn't result in NULL and empty string coming first, but it doesn't seem to.